include/tut/tut_assert.hpp

```diff
@@ -130,12 +139,22 @@ void ensure_equals(const M& msg, const LHS * const actual, const RHS * const exp
     }
 }
 
+#if __cplusplus >= 201103L
+template<typename M, typename LHS, typename RHS, typename EPS>
+typename std::enable_if<std::is_floating_point<LHS>::value || std::is_floating_point<RHS>::value, void>::type
+    ensure_equals(const M& msg, const LHS& actual, const RHS& expected, const EPS& epsilon)
+{
+	if((actual != expected)
+		&& (std::isnan(actual) || std::isnan(expected)
```

Comment:
We need to keep this tests, otherwise `std::abs()` will complain if it gets a NaN value.
We now have a consistent behavior with the standard: if one or both values are NaN, the test fails.